### PR TITLE
#299 enrich test data

### PR DESCRIPTION
**Pull request by UncleBu1728**
This PR introduces additional testing data by including calibration groups and interviews between users and dummy users. Please take a look at the pictures below for more details.

To ensure the correct group is returned when using the findGroups function in exclusive mode, a new function findGroupsByType was added. This function further filters the fields of ID to identify the correct group relationships accurately.

Future plans include adding more group relationships such as coaches and partnerships. Additionally, I plan to enrich the data with interview reviews for each test interview group.

image
image
image
image
image

### CODE REVIEW

File: tools/generateTestData.ts [hunk 1]
```diff
@@ -140,4 +145,93 @@ async function upsertSummary(groupId: string, transcriptId: string, startedAt: n
...
+      console.log(`Creating MentorInterview for [${users.map(u => u.name)}, ${tu.name}]`);
+      if ((await findGroupsByType("Interview", [tu.id, ...userIds])).length != 0) continue;
+      await createInterview("MentorInterview", calibrations.mentorCalibration.id, tu.id, [...userIds]);
+    };
+  }
+}
+
+// Checking fields of IDs to return matching exclusive groups
+// Calibration Groups are excluded since they do not have userIds
+async function findGroupsByType(groupType: "Group" | "Interview" | "Partnership" | "Coachee", userIds: string[]) {
```
Comment (weihanwang) [inline tools/generateTestData.ts:201]: "Unowned"

File: tools/generateTestData.ts [hunk 2]
```diff
@@ -140,4 +145,93 @@ async function upsertSummary(groupId: string, transcriptId: string, startedAt: n
...
+
+  console.log("Creating Test MentorIntervew Calibration");
+  const [mentorCalibration, mentorCalibrationCreated] = await Calibration.findOrCreate({
+    where: {
+      type: 'MentorInterview',
+      name: '导师测试面试组',
+    },
+    defaults: { active: true, }
+  });
+
```
Comment (weihanwang) [inline tools/generateTestData.ts:168]: delete creation of mentor interview.

File: tools/generateTestData.ts [hunk 3]
```diff
@@ -140,4 +145,93 @@ async function upsertSummary(groupId: string, transcriptId: string, startedAt: n
...
     summary
   });
+}
+
+async function findOrCreateCalibration() {
+  console.log("Creating Test MenteeIntervew Calibration");
+  const [menteeCalibration, menteeCalibrationCreated] = await Calibration.findOrCreate({
+    where: {
+      type: 'MenteeInterview',
+      name: '学生测试面试组',
```
Comment (weihanwang) [inline tools/generateTestData.ts:155]: “学生测试面试组” =》“面试组A”

**Comment by UncleBu1728** [comment]
Latest commit screenshots

image
image
image
image

**Review by weihanwang — APPROVED** [review]

**Pull request merged**